src/SkirtBrim.cpp

```diff
@@ -387,15 +387,22 @@ Polygons SkirtBrim::getFirstLayerOutline(const int extruder_nr /* = -1 */)
         const bool skirt_around_prime_tower_brim = storage.primeTower.enabled && global_settings.get<bool>("prime_tower_brim_enable");
         const bool include_prime_tower = ! skirt_around_prime_tower_brim; // include manually otherwise
 
-        const int skirt_height = global_settings.get<int>("skirt_height");
         first_layer_outline = Polygons();
-        for (int i_layer = layer_nr; i_layer <= skirt_height; ++i_layer)
+        int skirt_height = 0;
+        for (const auto& extruder : Application::getInstance().current_slice->scene.extruders)
         {
-            first_layer_outline =
-                first_layer_outline.unionPolygons
-                (
-                    storage.getLayerOutlines(i_layer, include_support, include_prime_tower, external_only, extruder_nr)
-                );
+            skirt_height = std::min(std::max(skirt_height, extruder.settings.get<int>("skirt_height")), static_cast<int>(storage.print_layer_count));
+        }
```

Comment:
I have a slight preference for something like this:
```suggestion
            skirt_height = std::max(skirt_height, extruder.settings.get<int>("skirt_height"));
        }
        skirt_height = std::min(skirt_height, static_cast<int>(storage.print_layer_count));
```
It is more readable in my opinion as you sequence the operations more. First you calculate the max skirt height of the various extrudes, then you cap this height to the layer count.